stdlib/private/StdlibCollectionUnittest/CheckRangeReplaceableCollectionType.swift

```diff
@@ -44,6 +51,78 @@ internal enum RangeSelection {
         return start..<end
     }
   }
+
+  internal func countableRange<C : Collection>(in c: C) -> CountableRange<C.Index> {
+    switch self {
+      case .emptyRange: return c.endIndex..<c.endIndex
+      case .leftEdge: return c.startIndex..<c.startIndex
+      case .rightEdge: return c.endIndex..<c.endIndex
+      case .middle:
+        let start = c.index(c.startIndex, offsetBy: c.count / 4)
+        let end = c.index(c.startIndex, offsetBy: 3 * c.count / 4 + 1)
+        return start..<end
+      case .leftHalf:
+        let start = c.startIndex
+        let end = c.index(start, offsetBy: c.count / 2)
+        return start..<end
+      case .rightHalf:
+        let start = c.index(c.startIndex, offsetBy: c.count / 2)
+        let end = c.endIndex
+        return start..<end
+      case .offsets(let offsets):
+        let start = c.index(c.startIndex, offsetBy: numericCast(offsets.lowerBound))
+        let end = c.index(c.startIndex, offsetBy: numericCast(offsets.upperBound))
+        return start..<end
+    }
+  }
+
+  internal func closedRange<C : Collection>(in c: C) -> ClosedRange<C.Index> {
+    switch self {
+      case .emptyRange: fatalError("Closed range cannot be empty")
+      case .leftEdge: return c.startIndex...c.startIndex
+      case .rightEdge: return c.endIndex...c.endIndex
+      case .middle:
+        let start = c.index(c.startIndex, offsetBy: c.count / 4)
+        let end = c.index(c.startIndex, offsetBy: 3 * c.count / 4 + 1)
+        return start...end
+      case .leftHalf:
+        let start = c.startIndex
+        let end = c.index(start, offsetBy: c.count / 2)
+        return start...end
+      case .rightHalf:
+        let start = c.index(c.startIndex, offsetBy: c.count / 2)
+        let end = c.endIndex
+        return start...end
+      case .offsets(let offsets):
+        let start = c.index(c.startIndex, offsetBy: numericCast(offsets.lowerBound))
+        let end = c.index(c.startIndex, offsetBy: numericCast(offsets.upperBound))
+        return start...end
+    }
```

Comment:
Does it work?  I wouldn't expect it to work because the test data is written assuming the upper bound is not included in the range.  For example, consider `.rightHalf`.  In `closedRange()` it includes `c.endIndex`.  Then we call `replaceSubrange` on such a range, implying that we would like to replace the element located at `endIndex`, but there is none.